src/plugins/cascade/src/main/js/core/Utils.js

```diff
@@ -9,36 +9,95 @@ define(
   ],
     function (Tools) {
       var getImageClassesForDropdown = function (editor, defaultClassList) {
-        return getClassesForDropdown(editor, defaultClassList, "image");
+        return getClassesForDropdown(editor, defaultClassList, "img");
       };
       var getLinkClassesForDropdown = function (editor, defaultClassList) {
-        return getClassesForDropdown(editor, defaultClassList, "link");
+        return getClassesForDropdown(editor, defaultClassList, "a");
       };
       var getClassesForDropdown = function (editor, defaultClassList, element) {
         var editorSettings = editor.settings;
-        var classList = [];
+        var selectedNode = editor.selection.getNode();
+        var nodeName = selectedNode.nodeName;
+
+        if (nodeName.toLowerCase() !== element) {
+          var elementNode = editor.dom.create(element, {});
+         // editor.selection.setNode(elementNode); //adds node to DOM
```

Comment:
Create didn't insert a node into the editor, but also wouldn't work as a `Node` parameter to [dom.is()](https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/#is) unless it's either appended to the current selection, or attempted to set the node in the selection object: `selection.setNode(newNode)` but those options insert the element (which we're trying to avoid)

Reply:
We worked through this by using the Formatter class and an empty element on creation.